### PR TITLE
graph: resolve paths in submodules

### LINKED ISSUE #217: resolve workflow paths in submodules
When a data product is imported from another renku repository, the lineage resolution does not properly resolve the workflow path. E.g.

```
*  dogs@68d0cf12 notebooks/01_features_run.ipynb
| *  dogs@dfd6a2ae .renku/vendors/platfofm/gitlab/Red-Team/dogs/data/outputs/features.pkl
|/
*    dogs@68d0cf12 .renku/workflow/4d0c1937e4d74f8d829177f46b9b69f1_papermill.cwl
|\
@ |  dogs@5d8fe701 notebooks/01_features.ipynb
 /
@  dogs:sigh_reborn@93176cee .renku/vendors/platform/gitlab/LAD/sigh_reborn/data/invest/full_dataset.csv
```

`.renku/workflow/4d0c1937e4d74f8d829177f46b9b69f1_papermill.cwl` should be `.renku/vendors/platfofm/gitlab/Red-Team/dogs/.renku/workflow/4d0c1937e4d74f8d829177f46b9b69f1_papermill.cwl`

### PR DESCRIPTION
**Pull request by jirikuncar**
(closes #217)

### CODE REVIEW

**Comment by ciyer** [comment]
Unless I have misunderstood, I think the lineage is still incorrect.

I have a test case where I import a file from another project:
```
*  toy@b0d31712 .renku/vendors/local/Users/ciyer/Documents/sdsc/play/toy/data/players.csv
*  toy@4cf25728 .renku/workflow/556da5035e184b9a942e1083fb842225_preprocess.cwl
|           (part of .renku/workflow/2db06b2660324f91b689d04212b3a6fc.cwl#steps/step_1)
@  toy@2937a456 data/fifa18/complete.csv
```

I am running the following version of renku-python:
91cfb06 (HEAD -> jiri/217-resolve-submodule-paths) graph: resolve paths in submodules

The same import gives this output:
```
*  toy@bb8d22c5 .renku/vendors/local/Users/ciyer/Documents/sdsc/play/toy/data/players.csv
*  toy@4cf25728 .renku/vendors/local/Users/ciyer/Documents/sdsc/play/toy/.renku/workflow/556da5035e184b9a942e1083fb842225_preprocess.cwl
|           (part of .renku/workflow/2db06b2660324f91b689d04212b3a6fc.cwl#steps/step_1)
@  toy@2937a456 .renku/vendors/local/Users/ciyer/Documents/sdsc/play/toy/data/fifa18/complete.csv
```

This line ```|           (part of .renku/workflow/2db06b2660324f91b689d04212b3a6fc.cwl#steps/step_1)``` seems incorrect. The workflow that is referenced is in the ```toy``` project, where I imported the data from, not in project of the cd.

File: renku/cli/_graph.py [hunk 1]
```diff
@@ -190,9 +190,17 @@ def add_workflow(self, commit, path, cwl=None, file_key=None):
             })
             step_map[step.id] = tool_key
 
-            self.G.nodes[tool_key]['workflow'] = workflow
-            self.G.nodes[tool_key]['workflow_path'
-                                   ] = path + '#steps/' + step.id
+            # The workflow path must be relative to the current tool since
+            # we might me inside a submodule.
```
Comment (ciyer) [inline renku/cli/_graph.py]: 'we might be' (not 'we might me')

File: renku/cli/_ascii.py [hunk 2]
```diff
@@ -88,15 +94,21 @@ def __iter__(self):
...
-        if workflow:
+        formatted_sha1 = _format_sha1(self.graph, node)
+        result = [formatted_sha1 + ' ' + self.graph._format_path(node[1])]
+        workflow_path = self.graph.G.nodes[node].get('workflow_path')
+        if workflow_path:
+            workflow_path = click.style(
+                self.graph._format_path(
+                    os.path.normpath(os.path.join(node[1], workflow_path))
+                ),
+                fg='blue'
```
Comment (ciyer) [inline renku/cli/_ascii.py]: Would be helpful to use a constant that describes the semantics of the color instead of specifying the color directly.

Comment (jirikuncar) [reply renku/cli/_ascii.py]: I will leave it for now and create an issue.

File: renku/cli/_graph.py (hunk 1 above)
Comment (jirikuncar) [reply renku/cli/_graph.py]: Thanks! I will fix it now.

**Pull request merged**